Linking error when cross compiling to aarch64-apple-darwin target

I tried cross-compiling rav1e from `x86_64-apple-darwin` to the `aarch64-apple-darwin` target (on Rustc 1.49.0-beta.2), and it failed while on linking rav1e. 

The full log can be found in this run [starting at line 92](https://github.com/EwoutH/rav1e/runs/1487102011#step:12:92) of the build step.
```
error: linking with `cc` failed: exit code: 1
  |
  = note: "cc" "-arch" "arm64" "-L" "/Users/runner/.rustup/toolchains/beta-x86_64-apple-darwin/lib/rustlib/aarch64-apple-darwin/lib" "/Users/runner/work/rav1e/rav1e/target/aarch64-apple-darwin/release/deps/rav1e.10utxkcv71e68b7c.rcgu.o" "/Users/runner/work/rav1e/rav1e/target/aarch64-apple-darwin/release/deps/rav1e.10yus5ksl057ygch.rcgu.o" "/Users/runner/work/rav1e/rav1e/target/aarch64-apple-darwin/release/deps/rav1e.11al8o6kw9fhx7gc.rcgu.o" 
....
```
And from [line 95](https://github.com/EwoutH/rav1e/runs/1487102011#step:12:95):
```
  = note: Undefined symbols for architecture arm64:
            "_rav1e_inv_txfm_add_flipadst_flipadst_4x4_16bpc_neon", referenced from:
                rav1e::asm::aarch64::transform::inverse::INV_TXFM_HBD_FNS::hbad123e9dd01a6df in rav1e.rav1e-2481e4f9553b2a72.icqhj6knk353rpc.rcgu.o.rcgu.o
            "_rav1e_inv_txfm_add_flipadst_dct_4x4_16bpc_neon", referenced from:
                rav1e::asm::aarch64::transform::inverse::INV_TXFM_HBD_FNS::hbad123e9dd01a6df in rav1e.rav1e-2481e4f9553b2a72.icqhj6knk353rpc.rcgu.o.rcgu.o
...
```
Exact details about the build environment [here](https://github.com/EwoutH/rav1e/runs/1487102011#step:1:1) and about the Rust installation and toolchain [here](https://github.com/EwoutH/rav1e/runs/1487102011#step:7:1).

--- a/build.rs
+++ b/build.rs
@@ -132,6 +132,9 @@ fn build_asm_files() {
 
   let dest_path = Path::new(&out_dir).join("config.h");
   let mut config_file = File::create(&dest_path).unwrap();
+  if env::var("CARGO_CFG_TARGET_OS").unwrap() == "macos" {
+    config_file.write(b" #define PREFIX 1\n").unwrap();
+  }
   config_file.write(b" #define PRIVATE_PREFIX rav1e_\n").unwrap();
   config_file.write(b" #define ARCH_AARCH64 1\n").unwrap();
   config_file.write(b" #define ARCH_ARM 0\n").unwrap();

